patches/api/0446-Add-FluidState-API.patch

```diff
@@ -0,0 +1,171 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: vicisacat <victor.branchu@gmail.com>
+Date: Fri, 17 Nov 2023 20:21:47 +0100
+Subject: [PATCH] Add FluidState API
+
+
+diff --git a/src/main/java/io/papermc/paper/block/fluid/FluidData.java b/src/main/java/io/papermc/paper/block/fluid/FluidData.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..ece4b71a5882d43f8149e559c746f4dba56648e1
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/block/fluid/FluidData.java
+@@ -0,0 +1,75 @@
++package io.papermc.paper.block.fluid;
++
++import io.papermc.paper.math.Position;
++import org.bukkit.Fluid;
++import org.bukkit.Location;
++import org.bukkit.World;
++import org.bukkit.util.Vector;
++import org.jetbrains.annotations.NotNull;
++
++public interface FluidData extends Cloneable {
```

Comment:
This was the original name in MachineMaker's PR. And I guess it makes sense, but shouldn't it be named differently to not confuse it with the minecraft one?

Reply:
Well…. We aren’t going to confuse it with the Minecraft name. Mayching MC names is ideal.

Reply:
I do not agree. This isn't the equivalent of a bukkit BlockState, which is what it will be confused with if we call it FluidState. It's the equivalent of an nms BlockState which the api calls BlockData. 

nms FluidState does not represent a block in the world, all the methods on it require a Level.

Reply:
Okay, that's fair. discard.